### ISSUE TITLE
Presence of invalid/incomplete certs prevents other valid certs from being made available to haproxy.

### ISSUE THREAD

**Issue by sangeethah** [rancher/rancher#8401]
Rancher Versions: Build from master
LB image - prachidamle/lb-service-haproxy:latest

Steps to reproduce the problem:
Created LB with the following cert directories for cert and default_cert , where /certs/mycerts directory had certs for test2.com and default_cert had test5.com.

test2.com and test5.com get loaded to the haproxy as expected.

Add domain - test3.com with only privkey.pem ( No fullchain.pem).
Notice that lbconfig reload reports error as expected.

Add domain - test4.com with both fullchain.pem  and privkey.pem
Notice that lbconfig reload reports the same error about  test3.com.

Certs dierctory of haproxy now has both test3.com and test4.com. But test4.com certificates are not actually being honored by Lb service.
 Not able to access Lb service using certificates of test4.com 

```
root@47b05bd898f1:/# ls /etc/haproxy/certs/current
test2.com.pem  test3.com.pem  test4.com.pem  test5.com.pem
root@47b05bd898f1:/#
```

Contents of certs directory:

```
total 8
lrwxrwxrwx 1 root root   13 Apr  3 21:07 fullchain.pem -> test4.com.crt
lrwxrwxrwx 1 root root   13 Apr  3 21:07 privkey.pem -> test4.com.key
-rw-r--r-- 1 root root 1164 Mar 31 00:46 test4.com.crt
-rw-r--r-- 1 root root 1679 Mar 31 00:46 test4.com.key

./test3.com:
total 8
lrwxrwxrwx 1 root root   13 Apr  3 21:05 privkey.pem -> test3.com.key
-rw-r--r-- 1 root root 1164 Mar 31 00:46 test3.com.crt
-rw-r--r-- 1 root root 1679 Mar 31 00:46 test3.com.key

./test2.com:
total 8
lrwxrwxrwx 1 root root   13 Apr  3 19:27 fullchain.pem -> test2.com.crt
lrwxrwxrwx 1 root root   13 Apr  3 19:27 privkey.pem -> test2.com.key
-rw-r--r-- 1 root root 1164 Apr  3 19:27 test2.com.crt
-rw-r--r-- 1 root root 1679 Apr  3 19:27 test2.com.key

```

In this case , we should be able to prevent loading test3.com certificate which is incomplete. This would allow for the valid test4.com certs to be applied to haproxy config successfully.

**Comment by sangeethah** [comment]
Tested with lb image - prachidamle/lb-service-haproxy:v0.1

When incomplete certs are present in cert directory , this does not prevent valid certs from being made available to haproxy.

**Comment by deniseschannon** [comment]
Official image released with rancher/lb-service-haproxy:v0.7.0 with the same fix.
